offset_manager_test.go

```diff
@@ -228,6 +228,33 @@ func TestPartitionOffsetManagerMarkOffset(t *testing.T) {
 	coordinator.Close()
 }
 
+func TestPartitionOffsetManagerMarkOffsetWithRetention(t *testing.T) {
```

Comment:
Instead of a sleep and a `coordinator.Returns` you should be able to call `coordinator.setHandler` and validate the request values in the callback.